go/internal/feast/server/http_server.go

```diff
@@ -158,6 +164,23 @@ func logWithSpanContext(span tracer.Span) zerolog.Logger {
 
 	return logger
 }
+
+/*
+*
+This function fixes the bug when the entity join key type is INT32.
+It typecasts the entity values to int32, as unmarshalled JSON values only return int64.
+*/
+func typecastToInt32Type(val *repeatedValue, entityType prototypes.ValueType_Enum) {
```

Comment:
So the reason I didn't make it generic is that repeatedValue is specifically used for storing JSON values for feature views and entities, which currently doesn't support any other data types other than string, int and bytes. I'll have to move this code into the toProto() function if we want to make it generic though, because toProto() iterates through this already.